File: src/main/java/io/bindernews/thegrackle/variables/Magic2Var.java
package io.bindernews.thegrackle.variables;

import com.evacipated.cardcrawl.modthespire.lib.SpireField;
import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
import com.megacrit.cardcrawl.cards.AbstractCard;
import io.bindernews.bnsts.IVariable;
import io.bindernews.thegrackle.GrackleMod;
import lombok.experimental.Accessors;

@Accessors(fluent = true)
public class Magic2Var extends AbstractSimpleVariable implements IVariable {
    public static final String KEY = GrackleMod.makeId("magic2");

    public static final Magic2Var inst = new Magic2Var();

    private Magic2Var() {
        super(Fields.magic2, new VariableInst(-1, -1, false));
    }

    @Override
    public String key() {
        return KEY;
    }


    @SpirePatch(clz = AbstractCard.class, method = SpirePatch.CLASS)
    public static class Fields {
        public static SpireField<VariableInst> magic2 = new SpireField<>(() -> null);
    }
}

File: src/main/kotlin/net/bindernews/grackle/variables/Magic2Var.kt
package net.bindernews.grackle.variables

import com.evacipated.cardcrawl.modthespire.lib.SpireField
import com.evacipated.cardcrawl.modthespire.lib.SpirePatch
import com.megacrit.cardcrawl.cards.AbstractCard
import net.bindernews.grackle.GrackleMod.Companion.MOD_ID

class Magic2Var private constructor() : AbstractSimpleVariable(Fields.grackleMagic2, VariableInst()) {
    override fun key(): String {
        return KEY
    }

    @SpirePatch(clz = AbstractCard::class, method = SpirePatch.CLASS)
    object Fields {
        @JvmField var grackleMagic2 = SpireField<VariableInst?> { null }
    }

    companion object {
        const val KEY = "$MOD_ID:magic2"
        @JvmField
        val inst = Magic2Var()
    }
}